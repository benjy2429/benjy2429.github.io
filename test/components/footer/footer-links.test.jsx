import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FooterLinks from '../../../src/components/footer/footer-links';

const internalLinksProps = {
  title: 'Menu',
  links: [
    {
      name: 'Link 1',
      href: '/test'
    }
  ]
};

const externalLinksProps = {
  title: 'Menu',
  links: [
    {
      name: 'Link 1',
      href: 'https://test.com'
    }
  ]
};

describe('Footer component', () => {
  it('renders a list of internal links', () => {
    const component = shallow(<FooterLinks {...internalLinksProps} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders a list of external links', () => {
    const component = shallow(<FooterLinks {...externalLinksProps} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
