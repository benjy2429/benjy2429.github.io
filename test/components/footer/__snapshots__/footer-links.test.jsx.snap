// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders a list of external links 1`] = `
<div>
  <h3
    className="footer-links-title"
  >
    Menu
  </h3>
  <ul
    className="footer-links"
  >
    <li
      key="Link 1"
    >
      <a
        className="link-light"
        href="https://test.com"
      >
        Link 1
      </a>
    </li>
  </ul>
</div>
`;

exports[`Footer component renders a list of internal links 1`] = `
<div>
  <h3
    className="footer-links-title"
  >
    Menu
  </h3>
  <ul
    className="footer-links"
  >
    <li
      key="Link 1"
    >
      <[object Object]
        className="link-light"
        to="/test"
      >
        Link 1
      </[object Object]>
    </li>
  </ul>
</div>
`;
